src/sentry/static/sentry/app/components/sampleTransitioner/sampleContainer.jsx

```diff
@@ -34,203 +19,197 @@ export class SampleContainer extends React.Component {
+    const { containerDirectionality } = this.props;
+    return containerDirectionality === SampleContainerDirectionality.TARGET;
   }
 
   isTransitionSource(location) {
-    const { transitionSources } = this.props;
+    const { transitionSourceLocations } = this.props;
 
     if (!this.isSourceContainer()) {
       return false;
     }
 
-    return !!transitionSources.find(tl => tl.equals(location));
+    return !!transitionSourceLocations.find(tl => tl.equals(location));
   }
 
   isActiveTransitionSource(location) {
-    const { activeSampleTransition } = this.props;
-    let isActiveTransitionSrc = false;
-
-    if (!this.isSourceContainer()) {
-      return isActiveTransitionSrc;
-    }
+    const { activeSampleTransitionSourceLocation } = this.props;
 
-    if (activeSampleTransition) {
-      const activeSampleTransitionSource = activeSampleTransition.getSource();
-      if (activeSampleTransitionSource) {
-        isActiveTransitionSrc = activeSampleTransitionSource.equals(location);
-      }
+    if (!this.isSourceContainer() || !activeSampleTransitionSourceLocation) {
+      return false;
     }
 
-    return isActiveTransitionSrc;
+    return  activeSampleTransitionSourceLocation.equals(location);
   }
 
   isTransitionTarget(location) {
-    const { transitionTargets } = this.props;
+    const { transitionTargetLocations } = this.props;
 
     if (!this.isTargetContainer()) {
       return false;
     }
 
-    return !!transitionTargets.find(tl => tl.equals(location));
+    return !!transitionTargetLocations.find(tl => tl.equals(location));
   }
 
   isTransitionTargetOfHoveredSample(location) {
-    const { transitionTargetsOfHoveredSample } = this.props;
+    const { transitionTargetLocationsOfHoveredSample } = this.props;
 
     if (!this.isTargetContainer()) {
       return false;
     }
 
-    return !!transitionTargetsOfHoveredSample.find(tl => tl.equals(location));
+    return !!transitionTargetLocationsOfHoveredSample.find(tl => tl.equals(location));
   }
 
-  createRows() {
-    let rows = [];
-    let colsHeader = [];
-    let key;
-
-    key = '-1_-1';
-    colsHeader.push(<td key={key} style={cellStyleHeader} />);
-    for (let c = 0; c < this.props.cols; c++) {
-      key = `${c}_-1`;
-      colsHeader.push(
-        <td key={key} style={this.getHeaderStyle(-1, c)}>
-          {this.getColIndicator(c)}
-        </td>
-      );
+  onMouseOut() {
+    const { onMouseOut } = this.props;
+
+    if (this.state.hoverRow || this.state.hoverCol) {
+      this.setState({hoverRow: null, hoverCol: null});
     }
-    rows.push(<tr>{colsHeader}</tr>);
-
-    for (let r = 0; r < this.props.rows; r++) {
-      let cols = [];
-      // Get all samples with this row
-      const rowSamples = this.props.samples.filter(s => s.location.row === r);
-
-      key = `-1_${r}`;
-      cols.push(
-        <td key={key} style={this.getHeaderStyle(r, -1)}>
-          {this.getRowIndicator(r)}
-        </td>
+
+    onMouseOut();
```

Comment:
Is this recursive call accidental?

Reply:
It's not recursive. (A recursive call would be this.onMouseOut()). What this does is invoke the additional onMouseOut function that has been passed in as a prop.